evennia/utils/evmenu.py

```diff
@@ -64,10 +64,63 @@ def another_node(caller, input_string, **kwargs):
 The return values must be given in the above order, but each can be
 returned as None as well. If the options are returned as None, the
 menu is immediately exited and the default "look" command is called.
+    text (str, tuple, dict or None): Text shown at this node. If a dict
+        it can also contain various formatting instructions. In such a
+        case the following keys are already supported:
+
+            - 'header' (dict): Dictionary of header data for the node
+            - 'text' (dict): Dictionary of text data for the node
+            - 'options' (dict): Dictionary of options data for the node
+            - 'footer' (dict): Dictionary of footer data for the node
+            - 'help' (dict): Dictionary of help data for the node
+
+            Each of these keys support their own dictionary with the
+            following keys:
+
+                - 'contents' (string or list): Required. String data for
+                    all keys other than 'options'. List in the case of
+                    options.
+                - 'formatter' (string): Optional. Designator of the
+                    function used to format the contents. Unless this
+                    code has been subclassed the only options are 'bars'
+                    or 'no_bars'.
+
+            In the case of 'options' the following additional keys are
+            supported by its dictionary:
+
+                - 'hidekeys' (list): A list of keys. The options for
+                    those keys will not be listed in the options table
+                    though the node will accept them as input.
+                - 'hidedescs' (list): As hidekey but using the desc of
+                    the options instead.
+                - 'movekeys' (list): A list of keys. The options for
+                    those keys will be put into a separate list after
+                    the options table.
+                - 'movedescs' (list): As movekeys but using the desc of
+                    the options instead.
+                - 'rows' (int): Will attempt to make the options table
+                    the specified number of rows.
```

Comment:
For consistency, this should be `formatter` like everywhere else, with optional kwargs being possible to send to it.

Reply:
My concern with using kwargs like that is that this is already a dict with sub-dicts. Adding kwargs to the formatter rather than leaving them outside as they are would make sub-sub-dicts.

Also, at that point why aren't things like ``contents`` part of the kwargs? I think perhaps the issue is I need to re-examing the explanation of the methodology. Essentially ``options`` is the instruction to format something ``formatter`` simply one of the kwargs about how to handle it (along with the contents, movekeys, etc.)

Reply:
Yes, it's a more complex return structure. It's still more consistent than these custom variable names IMO.